src/everest/detached/jobs/everest_server_api.py

```diff
@@ -0,0 +1,354 @@
+        )
+        .sign(key, hashes.SHA256(), default_backend())
+    )
+
+    # Write certificate and key to disk
+    makedirs_if_needed(cert_folder)
+    cert_path = os.path.join(cert_folder, cert_name + ".crt")
+    with open(cert_path, "wb") as f:
+        f.write(cert.public_bytes(serialization.Encoding.PEM))
+    key_path = os.path.join(cert_folder, cert_name + ".key")
+    pw = bytes(os.urandom(28))
+    with open(key_path, "wb") as f:
+        f.write(
+            key.private_bytes(
+                encoding=serialization.Encoding.PEM,
+                format=serialization.PrivateFormat.TraditionalOpenSSL,
+                encryption_algorithm=serialization.BestAvailableEncryption(pw),
+            )
+        )
+    return cert_path, key_path, pw
+
+
+def _sim_monitor(context_status, event=None, shared_data=None):
+    status = context_status["status"]
+    assert shared_data
+    shared_data[SIM_PROGRESS_ENDPOINT] = {
+        "batch_number": context_status["batch_number"],
+        "status": {
+            "running": status.get("Running", 0),
+            "waiting": status.get("Waiting", 0),
+            "pending": status.get("Pending", 0),
+            "complete": status.get("Finished", 0),
+            "failed": status.get("Failed", 0),
+        },
+        "progress": context_status["progress"],
+    }
+
+    if shared_data[STOP_ENDPOINT]:
+        return "stop_queue"
+
+
+def _opt_monitor(shared_data=None):
+    assert shared_data
+    if shared_data[STOP_ENDPOINT]:
+        return "stop_optimization"
+
+
+class ExperimentRunnerStatus(BaseModel):
+    status: Optional[str] = None
+    exit_code: Optional[EverestRunModelExitCode | OptimizerExitCode] = None
+    message: Optional[str] = None
+
+
+class ExperimentRunner(threading.Thread):
+    def __init__(self, everest_config, state: dict):
+        super().__init__()
+
+        self.everest_config = everest_config
+        self.state = state
+        self.status: Optional[ExperimentRunnerStatus] = None
+
+    def run(self):
+        run_model = EverestRunModel.create(
+            self.everest_config,
+            simulation_callback=partial(_sim_monitor, shared_data=self.state),
+            optimization_callback=partial(_opt_monitor, shared_data=self.state),
+        )
+
+        evaluator_server_config = EvaluatorServerConfig(
+            custom_port_range=range(49152, 51819)
+            if run_model.ert_config.queue_config.queue_system == QueueSystem.LOCAL
+            else None
+        )
+
+        try:
+            run_model.run_experiment(evaluator_server_config)
+            self.status = ExperimentRunnerStatus(
+                status="Experiment finished", exit_code=run_model.exit_code
+            )
+        except Exception:
+            self.status = ExperimentRunnerStatus(
+                status="Experiment failed", message=traceback.format_exc()
+            )
+
+    def get_status(self) -> Optional[ExperimentRunnerStatus]:
+        return self.status
```

Comment:
Should we consider using a background task for this instead? https://fastapi.tiangolo.com/tutorial/background-tasks/ Would be fairly similar, but can offload more to fastapi.

something like:
```
class ExperimentTask:
    def __init__(self, model: EverestRunModel ) -> None:
        self._model = model
        self.model_type = str(model.name())
        self._events: List[StatusEvents] = []

    def cancel(self) -> None:
        if self._model is not None:
            self._model.cancel()

    async def run(self):
        loop = asyncio.get_running_loop()
        logger.info(f"Starting experiment")

        port_range = None
        if self._model.queue_system == QueueSystem.LOCAL:
            port_range = range(49152, 51819)
        evaluator_server_config = EvaluatorServerConfig(custom_port_range=port_range)

        simulation_future = loop.run_in_executor(
            None,
            lambda: self._model.start_simulations_thread(
                evaluator_server_config
            ),
        )
```
then the endpoint can do:
```

@app.post("/experiments/", response_model=ExperimentOut)
async def submit_experiment(experiment: EverestConfig, background_tasks: BackgroundTasks):
    try:
        run_model = EverestRunModel.create(
            self.everest_config,
            simulation_callback=partial(_sim_monitor, shared_data=self.state),
            optimization_callback=partial(_opt_monitor, shared_data=self.state),
        )
    except ValueError as e:
        return HTTPException(
            status_code=420,
            detail="Experiment was not valid, failed with: {e}",
        )

    task = ExperimentTask(model=run_model)
    experiments[experiment_id] = task
    background_tasks.add_task(task.run)
    return ExperimentOut(id=uuid_here, type=task.model_type)
```

then the next step after this PR is to extract the events from the run model.